internal/cache/cache.go

```diff
@@ -10,27 +10,63 @@ import (
 // Cache provides a simple item store
 type Cache interface {
 	// Insert the string k
-	Insert(k string)
+	Insert(k string, v string)
 	// Exists proves the existence or lack-thereof of the item k in the cache
 	Exists(k string) (exists bool)
+	// Get returns the value of the key in the cache
+	Get(k string) (value string, ok bool)
+	// SetExpiration sets the expiration on the given item k without updating the value
+	SetExpiration(k string, t time.Duration)
 }
 
+const NoExpiration = gocache.NoExpiration
+
 type cache struct {
 	u *gocache.Cache
 }
 
-func (c *cache) Insert(k string) {
-	c.u.SetDefault(k, struct{}{})
+func (c *cache) SetExpiration(k string, t time.Duration) {
+	e, ok := c.u.Get(k)
+	if !ok {
+		return
+	}
+	c.u.Set(k, e, t)
+}
+
+func (c *cache) Insert(k string, v string) {
+	c.u.SetDefault(k, v)
 }
 
 func (c *cache) Exists(k string) (exists bool) {
 	_, exists = c.u.Get(k)
 	return exists
 }
 
+func (c *cache) Get(k string) (value string, ok bool) {
+	v, ok := c.u.Get(k)
+	if !ok {
+		return "", ok
+	}
+	value, ok = v.(string)
+	return value, ok
```

Comment:
this is overloading the meaning of the `ok` return value. I suggest adding a log warning here as it's a sign of potential funky behaviour

Reply:
Good call, I took it a step further and return a proper error since we have enough knowledge of how the cache is consumed to say this is a bad thing.